pkg/sql/parser/parse_test.go

```diff
@@ -1762,6 +1762,11 @@ func TestParse(t *testing.T) {
 		// {`CREATE CHANGEFEED FOR DATABASE foo INTO 'sink'`},
 		{`CREATE CHANGEFEED FOR TABLE foo INTO 'sink' WITH bar = 'baz'`},
 
+		{`CREATE REPLICATION STREAM FOR TENANT 1`},
+		{`CREATE REPLICATION STREAM FOR TENANT 1 WITH CURSOR='start'`},
+		{`CREATE REPLICATION STREAM FOR TENANT 1 INTO 'sink'`},
+		{`CREATE REPLICATION STREAM FOR TENANT 1 INTO 'sink' WITH CURSOR='start', DETACHED`},
```

Comment:
For fun, we could also add a test for non-tenant targets since that's supported by the grammar.